core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerFacade.java

```diff
@@ -104,18 +114,52 @@ public void loadVideo(Uri uri, ContentType contentType, ExoPlayerForwarder forwa
                 forwarder.mediaSourceEventListener()
         );
         exoPlayer.prepare(mediaSource, RESET_POSITION, DO_NOT_RESET_STATE);
+        setExoPlayerTextOutput(output);
     }
 
     public void selectAudioTrack(PlayerAudioTrack audioTrack) {
         // TODO check if we can read tracks from exoplayer directly
-        trackSelector.selectAudioTrack(audioTrack);
+        audioTrackSelector.selectAudioTrack(audioTrack);
     }
 
     public List<PlayerAudioTrack> getAudioTracks() {
-        return trackSelector.getAudioTracks();
+        return audioTrackSelector.getAudioTracks();
+    }
+
+    public void setSubtitleRendererOutput(TextRenderer.Output output) {
+        this.output = output;
+        setExoPlayerTextOutput(output);
+    }
+
+    private void setExoPlayerTextOutput(TextRenderer.Output output) {
+        if (exoPlayer == null) {
```

Comment:
You could call `showFirstAvailableSubtitleTrack` before loading a video as is a public method